src/managers/RoleManager.js

```diff
@@ -128,9 +130,10 @@ class RoleManager extends CachedManager {
    *   .catch(console.error);
    */
   async create(options = {}) {
-    let { name, color, hoist, permissions, position, mentionable, reason } = options;
+    let { name, color, hoist, permissions, position, mentionable, reason, icon } = options;
     if (color) color = resolveColor(color);
     if (typeof permissions !== 'undefined') permissions = new Permissions(permissions);
+    if (icon) icon = await DataResolver.resolveImage(icon);
```

Comment:
```suggestion
    icon &&= await DataResolver.resolveImage(icon);
```

Reply:
Shouldn't this also be done on the color line too?

Reply:
Shouldn't this also be done on the color line too?

Reply:
same as my above comment: This can be added for the whole lib in a dedicated PR for the sake of discussion and consistency